docs/winnt/Build.Cross-compile with minGW64 for Windows.md

```diff
@@ -0,0 +1,238 @@
+
+
+
+* **Build Platform**: this is the platform you are running on (for Linux this will be something like x86_64-pc-linux-gnu)
+ 
+* **Host Platform**: This is the platform you want the end product to run on, for this tutorial, the host will be x86_64-w64-mingw32
+
+* **Target Platform**:  This is the platform that what you are building will generate code for (generally only relevant when you are building a compiler) 
+
+Don't claim these are well picked names, they are what they are.
+
+Under Linux and similar platforms, autoconf provides most of the heavy
+lifting for cross-compilation that is needed. You just need to provide
+it the platforms triplet for the build platform and the host platform. Assuming your system is
+setup to match all its conventions, it will do all its normal magic
+using the proper flavors of everything from there.
+
+In theory, this means to compile an autoconf source based that targets
+Linux-like systems to Windows we just need to do:
+
+
+
+```
+./configure --host=x86_64-w64-mingw32
+```
+
+the build host will be detected, and the `Makefile` and needed configuration  files will be created using the need tools, and you'll just have to run `make` to build everything.  The caveat is that everything need to be in the "expected" locations for this to work. Autoconf will still do pretty much all the work for us, but it will take a bit of tweaking.
+
+---
+### Setting up the Cross-Build Environement
+
+
+Going through specifics, we going assume a base install of Debian Linux 11.x, bullseye, `x86_64`. Because it has (almost) everything needed in its packaging system, and because I have setup where I can  easily spin up a clean install of it. While package names may differ,  most major distributions will have same tools available.
+
+ See the necessary packages for a variety of platforms. <https://mxe.cc/#requirements> for solid list for most platforms. MXE is specifically for the task at hand. Unfortunately, while designed for this specific task, it's currently not quite friendly to use. Minimally, its current binary packages are very outdated. 
+
+For a solid build environment, we want to install these packages(we'll include tools for both 32-bit and 64-bit targets). This list definitely isn't minimal, but given a solid base to compiler most anything, including any additional compiling cross-compilers if they ever are needed. 
+
+```
+     apt install \\
+            autoconf automake autopoint             \\
+            bash bison build-essential bzip2        \\
+            flex                                    \\
+            g++ g++-multilib gettext git gperf      \\
+            intltool                                \\
+            libc6-dev-i386 libgdk-pixbuf2.0-dev     \\ 
+            libltdl-dev libssl-dev libtool-bin      \\
+            libxml-parser-perl                      \\
+            make openssl                            \\
+            patch perl python3 ruby                 \\
```

Comment:
Which part of this needs Ruby? What about the Perl and Python dependencies - where are they coming from?

Reply:
I pulled the prerequisites from MXE, there were what was needed to compile a whole pile of things from source for cross-compiling. Scrapped the list entirely, and worked from clean install + build essentials. Also checked if anything different would be need on Ubuntu (since GH CI stuff is setup on Ubuntu), and documented additions.